File: packages/python-sdk/chemvault/resources/documents.py
from __future__ import annotations

from typing import Any


class DocumentsResource:
    def __init__(self, client) -> None:
        self._client = client

    def upload(
        self,
        *,
        project_id: str,
        file_path: str,
        auto_parse: bool = True,
        auto_extract: bool = False,
    ):
        return self._client.upload_file(
            file_path,
            project_id=project_id,
            auto_parse=auto_parse,
            auto_extract=auto_extract,
        )

    def list(self, **params: Any):
        return self._client.request("GET", "/v1/documents", params=params)

    def retrieve(self, document_id: str):
        return self._client.request("GET", f"/v1/documents/{document_id}")

    def status(self, document_id: str):
        return self._client.request("GET", f"/v1/documents/{document_id}/status")

    def chunks(self, document_id: str):
        return self._client.request("GET", f"/v1/documents/{document_id}/chunks")

    def estimate(self, document_id: str):
        return self._client.request("POST", f"/v1/documents/{document_id}/estimate")

    def extract(self, document_id: str, *, mode: str = "standard", model: str | None = None):
        return self._client.request(
            "POST",
            f"/v1/documents/{document_id}/extract",
            json={"mode": mode, "model": model},
        )

    def records(self, document_id: str, *, include_unapproved: bool = False):
        return self._client.request(
            "GET",
            f"/v1/documents/{document_id}/records",
            params={"include_unapproved": include_unapproved},
        )

File: packages/js-sdk/src/resources/documents.ts
import type { ChemVault } from "../client.js";

export type DocumentUploadParams = {
  projectId: string;
  file: Blob | File;
  filename?: string;
  autoParse?: boolean;
  autoExtract?: boolean;
};

export class DocumentsResource {
  constructor(private readonly client: ChemVault) {}

  upload(params: DocumentUploadParams) {
    const formData = new FormData();
    formData.set("project_id", params.projectId);
    formData.set("auto_parse", String(params.autoParse ?? true));
    formData.set("auto_extract", String(params.autoExtract ?? false));
    if (params.filename) {
      formData.set("file", params.file, params.filename);
    } else {
      formData.set("file", params.file);
    }
    return this.client.request("POST", "/v1/documents", { formData });
  }

  list(params: Record<string, unknown> = {}) {
    return this.client.request("GET", "/v1/documents", { query: params });
  }

  retrieve(documentId: string) {
    return this.client.request("GET", `/v1/documents/${documentId}`);
  }

  status(documentId: string) {
    return this.client.request("GET", `/v1/documents/${documentId}/status`);
  }

  chunks(documentId: string) {
    return this.client.request("GET", `/v1/documents/${documentId}/chunks`);
  }

  estimate(documentId: string) {
    return this.client.request("POST", `/v1/documents/${documentId}/estimate`);
  }

  extract(documentId: string, params: { mode?: string; model?: string } = {}) {
    return this.client.request("POST", `/v1/documents/${documentId}/extract`, {
      body: { mode: params.mode ?? "standard", model: params.model },
    });
  }

  records(documentId: string, options: { includeUnapproved?: boolean } = {}) {
    return this.client.request("GET", `/v1/documents/${documentId}/records`, {
      query: { include_unapproved: options.includeUnapproved ?? false },
    });
  }
}
